revolv/payments/models.py

```diff
@@ -2,63 +2,231 @@
+    have the ability to "revoke" a repayment, if it was entered falsely.
+    Deleting an AdminRepayment also deletes any Repayment associated with it,
+    effectively erasing any trace of the repayment.
+
+    ::Signals::
+    pre_init
+        Make sure that related project is indeed complete, else throw a
+        ProjectNotCompleteException and to disallow instantiation of an invalid
+        AdminRepayment.
+    post_save
+        When an AdminRepayment is saved, a Repayment is generated for all
+        donors to a project, each weighed by that donor's proportion of the
+        contribution to the project.
+    """
+    amount = models.FloatField()
+    admin = models.ForeignKey(RevolvUserProfile)
+    project = models.ForeignKey("project.Project")
+
+    created_at = models.DateTimeField(auto_now_add=True)
+
+    objects = AdminRepaymentManager()
+    factories = ImportProxy("revolv.payments.factories", "AdminRepaymentFactories")
+
+
+class AdminReinvestmentManager(models.Manager):
+    """
+    Manager for AdminReinvestment.
+    """
+
+    def reinvestments(self, admin=None, project=None, queryset=None):
+        """
+        :return: AdminReinvestment associated with this admin and project.
+        """
+        if queryset is None:
+            queryset = super(AdminReinvestmentManager, self).get_queryset()
+        if admin:
+            queryset = queryset.filter(admin=admin).order_by('created_at')
+        if project:
+            queryset = queryset.filter(project=project).order_by('created_at')
+        return queryset
+
+
+class AdminReinvestment(models.Model):
+    """
+    Model representing a single, admin-controlled "contact point" for a
+    reinvestment to an ongoing RE-volv project.
+
+    RE-volv usually only reinvests into a project at its launch, but it is still
+    possible for an admin to put in a reinvestment at any time. Creating an
+    AdminReinvestment automatically pools money from users with a non-zero pool
+    of reinvestable money, prioritizing users that have a preference for the
+    Category of the project being reinvested into. (A user's pool of
+    reinvestable money consists of repayments to that user.) We generate a
+    Payment of type "reinvestment" for each user that we pooled money from with
+    the amount of money that we pooled from that user, and also decrement
+    that user's pool of reinvestable money.
+
+    An AdminReinvestment cannot be created if there are insufficient funds.
+
+    We need a single "contact point" representing a reinvestment so that admins
+    have the ability to "revoke" a reinvestment, if it was entered falsely.
+    Deleting an AdminReinvestment also deletes any reinvestment Payments
+    associated with it, effectively erasing any trace of the reinvestment.
+
+    ::Signals::
+    pre_init
+        Raises a NotEnoughFundingException before __init__ if there are not
+        enough funds for this AdminReinvestment.
+    post_save
```

Comment:
Yeah, there's a `created` kwarg, I'll use that. The same should be done for `AdminReinvestment`.